### ISSUE TITLE
jmespath uses an old, vulnerable version of jQuery

### ISSUE THREAD

**Issue by BitForger** [pinojs/pino-pretty#223]
`jmespath` is currently using jQuery 1.7.2 which is referenced specifically by [this vulnerability](https://nvd.nist.gov/vuln/detail/CVE-2012-6708) in 2012. Is it possible to move to a different utility for this? It appears jmespath is unmaintained.

**Comment by kibertoad** [comment]
PR would be welcome :)

**Comment by jsumners** [comment]
How is that CVE applicable to a local CLI tool?

**Comment by BitForger** [comment]
> How is that CVE applicable to a local CLI tool?

I'll preface with I'm not a hacker or pen-tester but from my opinion...

I agree that the vulnerability in the context of this code is basically void, but it does appear in a dependency graph and nearly every tool that seeks out and flags vulnerabilities is flagging that dependency.

This is mostly a paperwork issue for enterprises who are trying to make sure attack vectors are mitigated. If you are open to finding a suitable replacement I can certainly look for and create a PR for it.

**Comment by jsumners** [comment]
We are certainly open.

**Comment by mcollina** [comment]
I think a replacement could be https://jsonata.org/. Wdyt @jsumners?

**Comment by jsumners** [comment]
I haven't used jmespath in a long time, but I know it's used by Ansible. JSONata would be a complete query language change and result in a semver major requiring documentation updates and a clear warning with comparitive example in the release notes. 

I am okay with whatever works.

**Comment by BitForger** [comment]
Here is a [fork and TS implementation](https://github.com/nanoporetech/jmespath-ts) that might be usable. Other than that, is jsonpath the more standard one being used? I've seen mostly jsonpath and jq being used primarily in my personal experience.

**Comment by BitForger** [comment]
Sure. I'll start work on it tonight

**Comment by tilleps** [comment]
How is jmespath using jQuery?  I don't see any references to jQuery:
https://github.com/jmespath/jmespath.js/search?q=jquery
https://github.com/jmespath/jmespath.js/blob/master/package.json

**Comment by BitForger** [comment]
I believe it might be grunt has a dependency on it. I'm laid up sick right now, but I'll try to check it tonight.

**Comment by mcollina** [comment]
@BitForger jamespath has no runtime dependencies at all.

**Comment by BitForger** [comment]
I just went through and checked the whitesource report and it's going on about an html file that's distributed by the jmespath npm package. It just points at the node_modules/<dir> by default and I had to dig in a bit further to see that. I don't know how that html file is making it in there but it is.

Either way, less of an issue, but food for thought anyway since it appears the creator has been absent on the project.
